File: solutions/go/bird-watcher/1/bird_watcher.go
package birdwatcher

// TotalBirdCount return the total bird count by summing
// the individual day's counts.
func TotalBirdCount(birdsPerDay []int) int {
	total := 0
	for _, count := range birdsPerDay {
		total += count
	}

	return total
}

// BirdsInWeek returns the total bird count by summing
// only the items belonging to the given week.
func BirdsInWeek(birdsPerDay []int, week int) int {
	// 1 0 -> 8
	// 2 8 -> 15
	total := 0
	start := (7 * week) - 7
	end := start + 7
	for _, count := range birdsPerDay[start:end] {
		total += count
	}

	return total
}

// FixBirdCountLog returns the bird counts after correcting
// the bird counts for alternate days.
func FixBirdCountLog(birdsPerDay []int) []int {
	for i, count := range birdsPerDay {
		if i%2 == 0 {
			birdsPerDay[i] = count + 1
		}
	}

	return birdsPerDay
}

File: solutions/javascript/bird-watcher/3/bird-watcher.js
// @ts-check
//
// The line above enables type checking for this file. Various IDEs interpret
// the @ts-check directive. It will give you helpful autocompletion when
// implementing this exercise.

/**
 * Calculates the total bird count.
 *
 * @param {number[]} birdsPerDay
 * @returns {number} total bird count
 */
export function totalBirdCount(birdsPerDay) {
  let total = 0

  for (let i = 0; i < birdsPerDay.length; i++) {
    total = total + birdsPerDay[i]
  }

  return total
}

/**
 * Calculates the total number of birds seen in a specific week.
 *
 * @param {number[]} birdsPerDay
 * @param {number} week
 * @returns {number} birds counted in the given week
 */
export function birdsInWeek(birdsPerDay, week) {
  const weekStartIndex = 7 * (week - 1)
  let total = 0

  for (let i = weekStartIndex; i < weekStartIndex + 7; i++) {
    total = total + birdsPerDay[i]
  }

  return total
}

/**
 * Fixes the counting mistake by increasing the bird count
 * by one for every second day.
 *
 * @param {number[]} birdsPerDay
 * @returns {number[]} corrected bird count data
 */
export function fixBirdCountLog(birdsPerDay) {
  for (let index = 0; index < birdsPerDay.length; index = index + 2) {
    if(index % 2 === 0){
      birdsPerDay[index] = birdsPerDay[index] + 1
    }
  }
  return birdsPerDay
}
